integration-cli/docker_cli_run_test.go

```diff
@@ -2346,42 +2346,6 @@ func TestRunReuseBindVolumeThatIsSymlink(t *testing.T) {
 	logDone("run - can remount old bindmount volume")
 }
 
-func TestVolumesNoCopyData(t *testing.T) {
-	defer deleteImages("dataimage")
-	defer deleteAllContainers()
-	if _, err := buildImage("dataimage",
-		`FROM busybox
-		 RUN mkdir -p /foo
-		 RUN touch /foo/bar`,
-		true); err != nil {
-		t.Fatal(err)
-	}
-
-	cmd := exec.Command(dockerBinary, "run", "--name", "test", "-v", "/foo", "busybox")
-	if _, err := runCommand(cmd); err != nil {
-		t.Fatal(err)
-	}
-
-	cmd = exec.Command(dockerBinary, "run", "--volumes-from", "test", "dataimage", "ls", "-lh", "/foo/bar")
```

Comment:
@cpuguy83 so this should copy what's in dataimage:/foo into the volume. However, if you ran `docker run -v /foo busybox touch /foo/hello` instead of keeping the /foo empty, then the line i'm commenting on would not copy. Correct?
If that's the case, I'm against this PR :( It would bring more corner cases and is just a slight convenience not worth the maintainability effort.